manual/src/refman/extensions/effects.etex

```diff
@@ -0,0 +1,778 @@
+(Introduced in 5.0)
+
+\textit{Note: Effect handlers in OCaml 5.0 should be considered experimental.
+Effect handlers are exposed in the standard library as a thin wrapper
+around their implementations in the runtime. They are not supported as a
+language feature with new syntax. You can rely on them to build non-local
+control-flow abstractions such as user-level threading that do not expose
+the effect handler primitives to the user. Expect breaking changes in the
+future.}
+
+Effect handlers are a mechanism for modular programming with user-defined
+effects. Effect handlers allow the programmers to describe
+\textit{computations} that \textit{perform} effectful \textit{operations},
+whose meaning is described by \textit{handlers} that enclose the computations.
+Effect handlers are a generalization of exception handlers and enable non-local
+control-flow mechanisms such as resumable exceptions, lightweight threads,
+coroutines, generators and asynchronous I/O to be composably expressed. In this
+tutorial, we shall see how some of these mechanisms can be built using effect
+handlers.
+
+\subsection{s:effects-basics}{Basics}
+
+To understand the basics, let us define an effect (that is, an operation) that
+takes an integer argument and returns an integer result. We name this effect
+"Xchg".
+
+\begin{caml_example*}{verbatim}
+open Effect
+open Effect.Deep
+
+type _ Effect.t += Xchg: int -> int t
+let comp1 () = perform (Xchg 0) + perform (Xchg 1)
+\end{caml_example*}
+
+We declare the exchange effect "Xchg" by extending the pre-defined extensible
+variant type "Effect.t" with a new constructor "Xchg: int -> int t". The
+declaration may be intuitively read as ``the "Xchg" effect takes an integer
+parameter, and when this effect is performed, it returns an integer''. The
+computation "comp1" performs the effect twice using the "perform" primitive and
+returns their sum.
+
+We can handle the "Xchg" effect by implementing a handler that always returns
+the successor of the offered value:
+
+\begin{caml_example}{verbatim}
+try_with comp1 ()
+{ effc = fun (type a) (eff: a t) ->
+    match eff with
+    | Xchg n -> Some (fun (k: (a, _) continuation) ->
+        continue k (n+1))
+    | _ -> None }
+\end{caml_example}
+
+"try_with" runs the computation "comp1 ()" under an effect handler that handles
+the "Xchg" effect. As mentioned earlier, effect handlers are a generalization
+of exception handlers. Similar to exception handlers, when the computation
+performs the "Xchg" effect, the control jumps to the corresponding handler.
+However, unlike exception handlers, the handler is also provided with the
+delimited continuation "k", which represents the suspended computation between
+the point of "perform" and this handler.
```

Comment:
At this point, I would have expected an explanation of the type parameter `a` of the type `(a, _) continuation`.